runtime/tests/checker/access_test.go

```diff
@@ -2286,3 +2286,119 @@ func TestCheckInvalidRestrictiveAccessModifier(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckAccountAccess(t *testing.T) {
+
+	t.Parallel()
+
+	location1A := common.AddressLocation{
+		Address: common.BytesToAddress([]byte{0x1}),
+		Name:    "A",
+	}
+
+	location1B := common.AddressLocation{
+		// NOTE: same address as A
+		Address: common.BytesToAddress([]byte{0x1}),
+		Name:    "B",
+	}
+
+	location2B := common.AddressLocation{
+		// NOTE: different address from A
+		Address: common.BytesToAddress([]byte{0x2}),
+		Name:    "B",
+	}
+
+	const importingCode = `
+	  import A from 0x1
+
+      pub contract B {
+          pub fun use() {
+              let b = A.a
+          }
+      }
+	`
+
+	type testCase struct {
+		location         common.Location
+		accessModeChecks map[sema.AccessCheckMode]func(*testing.T, error)
+	}
+
+	tests := []testCase{
+		{
+			location: location1B,
+			accessModeChecks: map[sema.AccessCheckMode]func(*testing.T, error){
+				sema.AccessCheckModeStrict:                   expectSuccess,
+				sema.AccessCheckModeNotSpecifiedRestricted:   expectSuccess,
+				sema.AccessCheckModeNotSpecifiedUnrestricted: expectSuccess,
+				sema.AccessCheckModeNone:                     expectSuccess,
+			},
+		},
+		{
+			location: location2B,
+			accessModeChecks: map[sema.AccessCheckMode]func(*testing.T, error){
+				sema.AccessCheckModeStrict:                   expectInvalidAccessError,
+				sema.AccessCheckModeNotSpecifiedRestricted:   expectInvalidAccessError,
+				sema.AccessCheckModeNotSpecifiedUnrestricted: expectInvalidAccessError,
+				sema.AccessCheckModeNone:                     expectSuccess,
+			},
+		},
+	}
+
+	for _, variableKind := range ast.VariableKinds {
+
+		t.Run(variableKind.Name(), func(t *testing.T) {
+
+			importedChecker, err := ParseAndCheckWithOptions(t,
+				fmt.Sprintf(
+					`
+			          pub contract A {
+                          access(account) %s a: Int
```

Comment:
seems there are some tabs along with the whitespaces at this line

Reply:
Right, cleaned up in 8a1a6b0